docs/designs/preprocess_with_analysis.md

```diff
@@ -0,0 +1,165 @@
+# Preprocess with Analysis Result Design
+This document describes the design of preprocessing feature inputs using analysis results.
```

Comment:
This document describes the design about how to use analysis result while preprocessing feature inputs.